File: old_projects/src/project3/utility/Point.java
package project3.utility;

public class Point extends Tuple<Integer, Integer> {

    public Point(int f1, int f2) {
        super(f1, f2);
    }

    public Point[] get4Neighbours() {
        Point[] neighbours = new Point[4];
        neighbours[0] = new Point(f1+1, f2); // right
        neighbours[1] = new Point(f1-1, f2); // left
        neighbours[2] = new Point(f1, f2-1); // up
        neighbours[3] = new Point(f1, f2+1); // down

        return neighbours;
    }

    public Point[] get8Neighbours() {
        Point[] neighbours = new Point[8];
        neighbours[0] = new Point(f1+1, f2);    // E
        neighbours[1] = new Point(f1+1, f2+1);  // SE
        neighbours[2] = new Point(f1, f2+1);    // S
        neighbours[3] = new Point(f1-1, f2+1);  // SW
        neighbours[4] = new Point(f1-1, f2);    // W
        neighbours[5] = new Point(f1-1, f2-1);  // NW
        neighbours[6] = new Point(f1, f2-1);    // N
        neighbours[7] = new Point(f1+1, f2-1);  // NE

        return neighbours;
    }
}

File: Simulator/Utility/Point.py
class Point:

    f1 = 0
    f2 = 0

    def __init__(self, f1, f2) -> None:
        self.f1 = f1
        self.f2 = f2

    def get4Neighbours(self):
        neighbours = []
        neighbours.append(Point(self.f1 + 1, self.f2))
        neighbours.append(Point(self.f1 - 1, self.f2))
        neighbours.append(Point(self.f1, self.f2 - 1))
        neighbours.append(Point(self.f1, self.f2 + 1))

        return neighbours

    def get8Neighbours(self):
        neighbours = []
        neighbours.append(Point(self.f1 + 1, self.f2))
        neighbours.append(Point(self.f1 + 1, self.f2 + 1))
        neighbours.append(Point(self.f1, self.f2 + 1))
        neighbours.append(Point(self.f1 - 1, self.f2 + 1))
        neighbours.append(Point(self.f1 - 1, self.f2))
        neighbours.append(Point(self.f1 - 1, self.f2 - 1))
        neighbours.append(Point(self.f1, self.f2 - 1))
        neighbours.append(Point(self.f1 + 1, self.f2 - 1))
        return neighbours

    def __str__(self) -> str:
        return '<' + str(self.f1) + ',' + str(self.f2) + '>'

    def __eq__(self, __o: object) -> bool:
        if not isinstance(__o,Point):
            return False
        return self.f1 == __o.f1 and self.f2 == __o.f2
    
    def __hash__(self) -> int:
        return hash((self.f1,self.f2))


if __name__ == '__main__':
    a = Point(1, 1)
    for neighbour in a.get4Neighbours():
        print(neighbour)
    print('**********')
    for neighbour in a.get8Neighbours():
        print(neighbour)